include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -556,35 +558,168 @@ class cpp_dec_float
    }
 
  private:
-   cpp_dec_float(std::initializer_list<std::uint32_t> lst,
-                 const exponent_type e = 0,
-                 const bool n = false) : data(),
-                                         exp(e),
-                                         neg(n),
-                                         fpclass(cpp_dec_float_finite),
-                                         prec_elem(cpp_dec_float_elem_number)
-   {
-      std::copy(lst.begin(),
-                lst.begin() + (std::min)(std::int32_t(lst.size()), cpp_dec_float_elem_number),
-                data.begin());
-
-      std::fill(data.begin() + (std::min)(std::int32_t(lst.size()), cpp_dec_float_elem_number),
-                data.end(),
-                std::uint32_t(0u));
-   }
-
    static bool data_elem_is_non_zero_predicate(const std::uint32_t& d) { return (d != static_cast<std::uint32_t>(0u)); }
    static bool data_elem_is_non_nine_predicate(const std::uint32_t& d) { return (d != static_cast<std::uint32_t>(cpp_dec_float::cpp_dec_float_elem_mask - 1)); }
    static bool char_is_nonzero_predicate(const char& c) { return (c != static_cast<char>('0')); }
 
    void from_unsigned_long_long(const boost::ulong_long_type u);
 
-   int cmp_data(const array_type& vd) const;
+   template <typename InputIteratorTypeLeft,
+             typename InputIteratorTypeRight>
+   static int compare_ranges(InputIteratorTypeLeft  a,
+                             InputIteratorTypeRight b,
+                             const std::uint32_t    count = cpp_dec_float_elem_number);
+
+   static std::uint32_t eval_add_n(      std::uint32_t* r,
+                                   const std::uint32_t* u,
+                                   const std::uint32_t* v,
+                                   const std::int32_t   count);
 
+   static std::uint32_t eval_subtract_n(      std::uint32_t* r,
+                                        const std::uint32_t* u,
+                                        const std::uint32_t* v,
+                                        const std::int32_t   count);
+
+   static void eval_multiply_n_by_n_to_2n(      std::uint32_t* r,
+                                          const std::uint32_t* a,
+                                          const std::uint32_t* b,
+                                          const std::uint32_t  count);
+
+   #if 0
```

Comment:
Another nit-pick: I understand why this #if 0 is left in for future reference, but a comment to that effect might help us 10 years from now, by which point I fully expect my memory to be long gone south anyway :)